losantmqtt/device.py

```diff
@@ -156,19 +157,19 @@ def connect(self, blocking=True):
 
         self._looping = blocking
         self._initial_connect = True
-        self._mqtt_client = mqtt.Client(self._device_id)
+        self._mqtt_client = mqtt.Client(self._device_id, transport=self._transport)
         self._mqtt_client.username_pw_set(self._key, self._secret)
 
-        port = 1883
+        port = 80 if self._transport == "websockets" else 1883
         if self._secure:
             self._mqtt_client.tls_set(ROOT_CA_PATH)
-            port = 8883
+            port = 443 if self._transport == "websockets" else 8883
 
         LOGGER.debug("Connecting to Losant as %s", self._device_id)
         self._mqtt_client.on_connect = self._cb_client_connect
         self._mqtt_client.on_disconnect = self._cb_client_disconnect
         self._mqtt_client.message_callback_add(self._command_topic(), self._cb_client_command)
-        self._mqtt_client.connect(Device.mqtt_endpoint, port, 15)
+        self._mqtt_client.connect(Device.mqtt_endpoint, 8883, 15)
```

Comment:
Should this use the port variable defined above?

Reply:
yes. the power of code review wins again!